metric/bigtable/src/main/java/com/spotify/heroic/analytics/bigtable/BigtableAnalyticsModule.java

```diff
@@ -87,13 +91,14 @@ public AnalyticsReporter reporter(final HeroicReporter reporter) {
     public Managed<BigtableConnection> connection(
         final AsyncFramework async, final ExecutorService executorService
     ) {
+        Optional<Integer> batchSize = empty();
```

Comment:
This was only about inlining the variable, instead of declaring it separately from where it is being used:

```java
Optional<Integer> batchSize = empty();
/* ... */
new BigtableConnectionBuilder(/*  */, batchSize)
```

Into:

```java
new BigtableConnectionBuilder(/*  */, empty())
```

I think you are already falling back to the default value (without accessing the constant) by using the the builder, so that should not be a problem?

Speaking of which, the static import for `BIGTABLE_BULK_MAX_ROW_KEY_COUNT_DEFAULT` never seems to be used?
It is also worth keeping in mind that `javac` will inline constants, so it is recommended against sharing them across libraries:
http://stackoverflow.com/questions/3524150/is-it-possible-to-disable-javacs-inlining-of-static-final-variables